### PR TITLE
chore: cicd service principal integration

### PR DESCRIPTION
**Pull request by IvanJobs**
Replace username/password with service principal login for Azure in CI/CD.

To finish: https://dev.azure.com/msazure/Microsoft%20Teams%20Extensibility/_workitems/edit/12421108

On hold for smoke test after cli's ready, but you can review it first.

### CODE REVIEW

File: docs/cicd_insider/README.md [hunk 1]
```diff
@@ -207,9 +199,43 @@ The scripts are based on a cross-platform TeamsFx command line tool [TeamsFx-CLI
...
+2. Assign a role to your Azure AD application to access your Azure subscription, and `Contributor` role is recommended. 
+3. Create a new Azure AD application secret.
+4. Grab your tenant id, application id(AZURE_SERVICE_PRINCIPAL_NAME), and the secret(AZURE_SERVICE_PRINCIPAL_PASSWORD) for use.
+
+For detailed guidelines, refer to [the official document](https://docs.microsoft.com/en-us/azure/active-directory/develop/howto-create-service-principal-portal). There're three ways to create service principal, [Azure portal](https://docs.microsoft.com/en-us/azure/active-directory/develop/howto-create-service-principal-portal), [PowerShell](https://docs.microsoft.com/en-us/azure/active-directory/develop/howto-authenticate-service-principal-powershell), [Azure CLI](https://docs.microsoft.com/en-us/cli/azure/create-an-azure-service-principal-azure-cli), and you can choose the way you like.
+
+# How to publish Teams app using Teams dev portal? 
+If there's any changes related to Teams app's manifest file, you may want to publish the Teams app again to update the manifest.
+
+To publish Teams app manually, you may leverage [Teams' dev portal](https://dev.teams.microsoft.com/home).
```
Comment (nliu-ms) [inline docs/cicd_insider/README.md:229]: It would be better to use `Developer Portal for Teams`

**Review by nliu-ms — APPROVED** [review]

File: docs/cicd_insider/README.md (hunk 1 above)
Comment (IvanJobs) [reply docs/cicd_insider/README.md:229]: agree, update it soon.

**Review by MuyangAmigo — APPROVED** [review]
Thanks, the documentation part looks good!

File: docs/cicd_insider/README.md [hunk 2]
```diff
@@ -116,12 +114,10 @@ steps:
   
 - task: Bash@3
   env:
-    AZURE_ACCOUNT_NAME: $(AZURE_ACCOUNT_NAME)
-    AZURE_ACCOUNT_PASSWORD: $(AZURE_ACCOUNT_PASSWORD)
+    SP_NAME: $(AZURE_SERVICE_PRINCIPAL_NAME)
+    SP_PASSWORD: $(AZURE_SERVICE_PRINCIPAL_PASSWORD)
+    TENANT_ID: $(AZURE_TENANT_ID)
```
Comment (MuyangAmigo) [inline docs/cicd_insider/README.md:119]: This should be `TENANT_ID` or `AZURE_TENANT_ID`?

**Review by eriolchan — APPROVED** [review]

**Pull request merged**